dali/test/python/test_python_function_operator.py

```diff
@@ -0,0 +1,147 @@
+
+    def define_graph(self):
+        images, labels = self.load()
+        processed = self.python_function(images)
+        return processed
+
+
+class FlippingPipeline(CommonPipeline):
+    def __init__(self, batch_size, num_threads, device_id, seed, image_dir):
+        super(FlippingPipeline, self).__init__(batch_size, num_threads, device_id, seed, image_dir)
+        self.flip = ops.Flip(horizontal=1)
+
+    def define_graph(self):
+        images, labels = self.load()
+        flipped = self.flip(images)
+        return flipped
+
+
+DEVICE_ID = 0
+
+
+def random_seed():
+    return int(random.random() * (1 << 32))
+
+
+def make_parser():
+    _parser = argparse.ArgumentParser(description='Python function operator test')
+    _parser.add_argument(
+        '-i', '--iters', default=32, type=int, metavar='N',
+        help='number of iterations to run (default: whole dataset)')
+    _parser.add_argument(
+        '-s', '--seed', default=random_seed(), type=int, metavar='N',
+        help='seed for random ops (default: random seed)')
+    _parser.add_argument(
+        '-w', '--num_workers', default=3, type=int, metavar='N',
+        help='number of worker threads (default: %(default)s)')
+    _parser.add_argument(
+        '-b', '--batch_size', default=1, type=int, metavar='N',
+        help='image batch size (default: %(default)s)')
+    return _parser
+
+
+def run_test(func, name):
+    pipe = BasicPipeline(args.batch_size, args.num_workers, DEVICE_ID, args.seed, images_dirs[0])
+    pyfunc_pipe = PythonOperatorPipeline(args.batch_size, args.num_workers, DEVICE_ID, args.seed, images_dirs[0], func)
+    pipe.build()
+    pyfunc_pipe.build()
+    print('test: ' + name)
+    for it in range(args.iters):
+        preprocessed_output, = pipe.run()
+        output, = pyfunc_pipe.run()
+        for i in range(len(output)):
+            assert numpy.array_equal(output.at(i), func(preprocessed_output.at(i)))
+    print('done')
+
+
+def one_channel_normalize(image):
+    return image[:, :, 1] / 255.
+
+
+def channels_mean(image):
+    r = numpy.mean(image[:, :, 0])
+    g = numpy.mean(image[:, :, 1])
+    b = numpy.mean(image[:, :, 2])
+    return numpy.array([r, g, b])
+
+
+def bias(image):
+    return numpy.array(image > 127, dtype=numpy.bool)
+
+
+def flip(image):
+    return numpy.fliplr(image)
+
+
+def flipping_test(args):
+    dali_flip = FlippingPipeline(args.batch_size, args.num_workers, DEVICE_ID, args.seed, images_dirs[0])
+    numpy_flip = PythonOperatorPipeline(args.batch_size, args.num_workers, DEVICE_ID, args.seed, images_dirs[0], flip)
+    dali_flip.build()
+    numpy_flip.build()
+    print('test: flip')
+    for it in range(args.iters):
+        numpy_output, = numpy_flip.run()
+        dali_output, = dali_flip.run()
+        for i in range(len(numpy_output)):
+            assert numpy.array_equal(numpy_output.at(i), dali_output.at(i))
+    print('done')
+
+
+if __name__ == '__main__':
```

Comment:
I think we could stick to how it's done in `test_pipeline.py` and such. Just name your testcases like `test_*`, remove this `main` section and then just invoke the tests with `nosetests`